estate/models/estate_property.py

```diff
@@ -0,0 +1,153 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+from datetime import date
+from dateutil.relativedelta import relativedelta
+from odoo.tools.float_utils import float_compare, float_is_zero
+from odoo.exceptions import ValidationError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+
+    _sql_constraints = [
+        (
+            "expected_price_strictly_positive",
+            "CHECK(expected_price > 0)",
+            "Expected price must be strictly positive",
+        ),
+        (
+            "selling_price_positive",
+            "CHECK(selling_price >= 0)",
+            "Selling price must be positive",
+        ),
+    ]
+
+    @api.constrains("expected_price", "selling_price")
+    def _check_selling_price(self):
+        for record in self:
+            if not float_is_zero(record.selling_price, precision_digits=2):
+                if (
+                    float_compare(
+                        record.selling_price,
+                        record.expected_price * 0.9,
+                        precision_digits=2,
+                    )
+                    < 0
+                ):
+                    raise ValidationError(
+                        "The selling price cannot be lower than 90% of the expected price!"
+                    )
+
+    def action_sold(self):
+        for record in self:
+            if record.state != "cancelled":
+                record.state = "sold"
+                return True
+            raise UserError("A cancelled property cannot be set as sold")
+
+    def action_cancel(self):
+        for record in self:
+            if record.state != "sold":
+                record.state = "cancelled"
+                return True
+            raise UserError("A sold property cannot be set as cancelled")
+
+    @api.depends("offer_ids.status")
+    def _compute_infos_from_accepted_offer(self):
+        for record in self:
+            for offer in record.offer_ids:
+                if offer.status == "accepted":
+                    record.buyer_id = offer.partner_id
+                    record.selling_price = offer.price
+                    return
+            record.buyer_id = None
+            record.selling_price = None
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_offer(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_offer = max(map(lambda r: r.price, record.offer_ids))
+            else:
+                record.best_offer = None
```

Comment:
The `.mapped` method does the mapping you did
```suggestion
            record.best_offer = max(record.offer_ids.mapped('price'), default=0.0)
```